rsfbclient-native/src/params.rs

```diff
@@ -119,10 +122,13 @@ impl ParamBuffer {
 
         let mut buffer = match info {
             Param::Text(s) => {
-                if s.len() > MAX_TEXT_LENGTH {
-                    binary_to_blob(s.into_bytes(), db, tr, ibase)?
+                let len = s.len();
+                let bytes = charset.encode(s)?;
```

Comment:
The length is wrong, you need to use the length after encoding, not before.

```suggestion
                let bytes = charset.encode(s)?;
                let len = bytes.len();
```

Reply:
Fixed!